freqtrade/plugins/pairlist/PercentVolumeChangePairList.py

```diff
@@ -0,0 +1,319 @@
+        if self._use_range:
+            filtered_tickers: List[Dict[str, Any]] = [{"symbol": k} for k in pairlist]
+
+            # get lookback period in ms, for exchange ohlcv fetch
+            since_ms = (
+                int(
+                    timeframe_to_prev_date(
+                        self._lookback_timeframe,
+                        dt_now()
+                        + timedelta(
+                            minutes=-(self._lookback_period * self._tf_in_min) - self._tf_in_min
+                        ),
+                    ).timestamp()
+                )
+                * 1000
+            )
+
+            to_ms = (
+                int(
+                    timeframe_to_prev_date(
+                        self._lookback_timeframe, dt_now() - timedelta(minutes=self._tf_in_min)
+                    ).timestamp()
+                )
+                * 1000
+            )
+
+            # todo: utc date output for starting date
+            self.log_once(
+                f"Using change range of {self._lookback_period} candles, timeframe: "
+                f"{self._lookback_timeframe}, starting from {format_ms_time(since_ms)} "
+                f"till {format_ms_time(to_ms)}",
+                logger.info,
+            )
+            needed_pairs: ListPairsWithTimeframes = [
+                (p, self._lookback_timeframe, self._def_candletype)
+                for p in [s["symbol"] for s in filtered_tickers]
+                if p not in self._pair_cache
+            ]
+
+            candles = self._exchange.refresh_ohlcv_with_cache(needed_pairs, since_ms)
+
+            for i, p in enumerate(filtered_tickers):
+                pair_candles = (
+                    candles[(p["symbol"], self._lookback_timeframe, self._def_candletype)]
+                    if (p["symbol"], self._lookback_timeframe, self._def_candletype) in candles
+                    else None
+                )
+
+                # in case of candle data calculate typical price and change for candle
+                if pair_candles is not None and not pair_candles.empty:
+                    pair_candles["rolling_volume_sum"] = (
+                        pair_candles["volume"].rolling(window=self._lookback_period).sum()
+                    )
+                    pair_candles["rolling_volume_change"] = (
+                        pair_candles["rolling_volume_sum"].pct_change() * 100
+                    )
+
+                    # ensure that a rolling sum over the lookback_period is built
+                    # if pair_candles contains more candles than lookback_period
+                    rolling_volume_change = pair_candles["rolling_volume_change"].fillna(0).iloc[-1]
+
+                    # replace change with a range change sum calculated above
+                    filtered_tickers[i]["rolling_volume_change"] = rolling_volume_change
+                    self.log_once(f"ticker {filtered_tickers[i]}", logger.info)
```

Comment:
I'm not sure if this pairlist will  give significantly different values from the regular `VolumePairlist` - and if it does - it should be incorporated into the VolumePairlist as a separate mode / sortkey - as it's not "that" different (though i'm not sure what benefit this would bring - so that should be discussed first if you think "this" is a pairlist we need).

---

it's however not what #10348 asked for - but it's "pairs with the biggest volume changes"

The issue however asking for the "top 20 assets with the highest increase within 24 hours" - but not in Volume - but in price.
the output would essentially, it'd be the "top gainers" list on binance (not sure where it's on the website - but on mobile it's on the first page).

It's essentially a pairlist that's probably VERY similar to `RangeStabilityFilter` - but sorting on "positive change" - whereas the `RangeStabilityFilter` just removes pairs (without reordering).

Reply:
okay i see. 
I see in the ticker structure we have `percentage` which provides the 24 hour percent price change in most of the exchanges.
**Binance:** 
`"priceChangePercent": "-0.0162",  //24-hour percent price change`
**ByBit:**
`price24hPcnt | string | Percentage change of market price relative to 24h`
Also i see the same is used in ccxt as:
`percentage = self.safe_string(ticker, 'price_change_percent_24h')` And `'percentage': self.safe_string(ticker, 'priceChangePercent'),`

Would it make sense to use it from tickers? That would been making `needstickers` **true**.  It would be a much simpler Filter where i won't be generating any pair list to start with and just sort and filter the PAIRs based on price changed over last 24hrs. And then same can later be extended to support candles and calculate price change over the `lookback_period` as mentiond in #10415 ?
What do you think?

Reply:
from tickers will always be a lot faster - simply because that's "1" call that can be shared, instead of one call per pair (and potentially more).

that's how volumePairlist works if supported (and if not configured for individual lookback_periods or timeframes.

In the end, it's "your" PR - so I'm happy with either having it - or removing all traces of it  and go with "tickers only"- but i don't think the current code is "that far" away from this - as it's "only" the percentage calculation that's missing - which you can probably "steal" from either RangeStabilityFilter  or VolatilityFilter ...

Reply:
Updated the calculations.

Renamed the PairList to `PercentChangePairList`. Made it similar to `VolumePairList` i.e. if `lookback_period` and `timeframe` provided, we use the candles to calculate the percent change. If not we use the tickers. Also added a `_ft_has.tickers_have_percentage` property to check support for each exchange. Made it True by default as `percentage` is part of ticker structure and additional check on **None** value is present.

Lookback Period Calculation using candles:
```python
   current_close = pair_candles["close"].iloc[-1]
   previous_close = pair_candles["close"].shift(self._lookback_period).iloc[-1]
   pct_change = (
     ((current_close - previous_close) / previous_close) if previous_close > 0 else 0
   )
```